src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerations.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines.AvoidMultipleEnumerations.FlowAnalysis;
+using static Microsoft.CodeQuality.Analyzers.MicrosoftCodeQualityAnalyzersResources;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines.AvoidMultipleEnumerations
+{
+    public abstract partial class AvoidMultipleEnumerations : DiagnosticAnalyzer
+    {
+        private const string RuleId = "CA1851";
+
+        private static readonly DiagnosticDescriptor MultipleEnumerableDescriptor = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(AvoidMultipleEnumerationsTitle)),
+            CreateLocalizableResourceString(nameof(AvoidMultipleEnumerationsMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.Disabled,
+            description: null,
+            isPortedFxCopRule: false,
+            isDataflowRule: true);
```

Comment:
Following assert will fire when we enable this rule: https://github.com/dotnet/roslyn-analyzers/blob/f3bdbd1e722f2c56ff85d702e78e3ffe070e38bd/src/Utilities/Compiler/DiagnosticDescriptorHelper.cs#L27-L28

We should possibly modify that assert to explicitly skip `CA1851` for that assert.

Reply:
Also found out that `isDataflowRule: false` passed to all descriptors in `PlatformCompatAnalyzer` is incorrect: https://github.com/dotnet/roslyn-analyzers/blob/f3bdbd1e722f2c56ff85d702e78e3ffe070e38bd/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs#L67

@buyaa-n We should update the PlatformCompatAnalyzer descriptors to use `isDataflowRule: true` and update the above assert to skip `CA1416` as an intentionally enabled by default dataflow rule in NetAnalyzers package.

Reply:
: ) I will leave this comment open